hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/AbstractEmbeddableMapping.java

```diff
@@ -110,7 +111,12 @@ public void setValues(Object component, Object[] values) {
 		}
 		else {
 			for ( int i = 0; i < values.length; i++ ) {
-				getAttributeMapping( i ).getPropertyAccess().getSetter().set( component, values[i] );
+				final Setter setter = getAttributeMapping( i ).getPropertyAccess().getSetter();
+				// We need to check the setter is not null,
+				// i.e.  PropertyAccessCompositeUserTypeImpl#getSetter() returns null;
+				if ( setter != null ) {
```

Comment:
Shouldn't we rather check outside of the loop if the `getRepresentationStrategy()` is someting custom and skip the whole loop altogether then? Also, it's a bit concerning that this method is called in the first place. Shouldn't we try to create a copy of the component instead?

Reply:
`TypeHelper#replaceAssociations()`, calls `ComnponentType#setPropertyValues`,
probably a better solution is to change this last method to :
```
public void setPropertyValues(Object component, Object[] values) {
    if ( compositeUserType != null ) {
	return;
    }
    mappingModelPart.getEmbeddableTypeDescriptor().setValues( component, values );
}
```
what do you think?

Reply:
I don't know if `TypeHelper#replaceAssociations()` will actually "do something", but maybe it's better to create a new copy of the component and replace that in the respective context if `compositeUserType != null`

Reply:
it should already be a copy, from the `TypeHelper#replaceAssociations` doc
```
* If the corresponding type is a component type, then apply {@link Type#replace} across the component
* subtypes but do not replace the component value itself.
```

Reply:
working on a solution